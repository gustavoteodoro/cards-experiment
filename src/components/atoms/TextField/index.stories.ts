import TextField from './index.vue';

export default { title: 'Atoms/TextField' };

export const pure = () => ({
  components: { TextField },
  template: '<text-field />'
});

export const withLabel = () => ({
  components: { TextField },
  template: '<text-field label="Label example" />'
});

export const withPlaceholder = () => ({
  components: { TextField },
  template: '<text-field label="Label example" placeholder="Placeholder example..." />'
});

export const withMaxLenght = () => ({
  components: { TextField },
  template: '<text-field label="Label example" maxlength="2" placeholder="Placeholder example..." />'
});

export const withMinLenght = () => ({
  components: { TextField },
  template: '<text-field label="Label example" minlength="2" placeholder="Placeholder example..." />'
});
