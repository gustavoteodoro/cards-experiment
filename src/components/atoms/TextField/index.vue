<template>
  <label :aria-label="[[ label ]]">
    <span>{{ label }}</span>
    <input
      type="text"
      :minlength="[[ minlength ]]"
      :maxlength="[[ maxlength ]]"
      :aria-placeholder="[[ placeholder ]]"
      :placeholder="[[ placeholder ]]"
      required
    />
  </label>
</template>

<script>
export default {
  name: 'TextField',
  props: {
    label: String,
    minlength: Number,
    maxlength: Number,
    placeholder: String,
    required: Boolean,
  },
};
</script>

<style scoped>
label {
  text-align: left;
}
span {
  display: block;
  font-size: 12px;
  margin-bottom: 4px;
}
input {
  display: block;
  padding: 9px 15px;
  font-size: 16px;
  border-radius: 5px;
  border: 2px solid #cfd2df;
}
</style>
