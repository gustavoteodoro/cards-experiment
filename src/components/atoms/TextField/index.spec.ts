import { mount } from "@vue/test-utils";
import TextField from "./index.vue"

describe("TextField", () => {
  it('renders correctly', () => {
    const wrapper = mount(TextField)
    expect(wrapper.element).toMatchSnapshot()
  })
})