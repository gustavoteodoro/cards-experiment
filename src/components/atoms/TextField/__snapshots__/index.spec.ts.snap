// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<label
  aria-label=""
>
  <span>
    
  </span>
   
  <input
    aria-placeholder=""
    maxlength=""
    minlength=""
    placeholder=""
    required="required"
    type="text"
  />
</label>
`;
